storm-webapp/src/main/java/org/apache/storm/daemon/ui/resources/StormApiResource.java

```diff
@@ -0,0 +1,702 @@
+
+    public static Meter getOwnerResourceSummariesMeter =
+            StormMetricsRegistry.registerMeter("ui:num-get-owner-resource-summaries-http-request");
+
+
+    /**
+     * /api/v1/cluster/configuration -> nimbus configuration.
+     */
+
+    @GET
+    @Path("/cluster/configuration")
+    @Produces("application/json")
+    public Response getClusterConfiguration(@QueryParam("callback") String callback) throws TException {
+        clusterConfigurationRequestMeter.mark();
+        try (NimbusClient nimbusClient = NimbusClient.getConfiguredClient(config)) {
+            return UIHelpers.makeStandardResponse(
+                    nimbusClient.getClient().getNimbusConf(),
+                    callback, false
+            );
+        }
+    }
+
+    /**
+     * /api/v1/cluster/summary -> cluster summary.
+     */
+    @GET
+    @Path("/cluster/summary")
+    @NimbusOp("getClusterInfo")
+    @Produces("application/json")
+    public Response getClusterSummary(@QueryParam("callback") String callback) throws TException {
+        clusterSummaryRequestMeter.mark();
+        try (NimbusClient nimbusClient = NimbusClient.getConfiguredClient(config)) {
+            String user = servletRequest.getRemoteUser();
+            return UIHelpers.makeStandardResponse(
+                    UIHelpers.getClusterSummary(
+                            nimbusClient.getClient().getClusterInfo(), user, config),
+                    callback
+            );
+        }
+    }
+
+    /**
+     * /api/v1/nimbus/summary -> nimbus summary.
+     */
+    @GET
+    @Path("/nimbus/summary")
+    @NimbusOp("getClusterInfo")
+    @Produces("application/json")
+    public Response getNimbusSummary(@QueryParam("callback") String callback) throws TException {
+        nimbusSummaryRequestMeter.mark();
+        try (NimbusClient nimbusClient = NimbusClient.getConfiguredClient(config)) {
+            return UIHelpers.makeStandardResponse(
+                    UIHelpers.getNimbusSummary(
+                            nimbusClient.getClient().getClusterInfo(), config),
+                    callback
+            );
+        }
+    }
+
+    /**
+     * /api/v1/owner-resources -> owner resources.
+     */
+    @GET
+    @Path("/owner-resources")
+    @NimbusOp("getOwnerResourceSummaries")
+    @Produces("application/json")
+    public Response getOwnerResources(@QueryParam("callback") String callback) throws TException {
+        getOwnerResourceSummariesMeter.mark();
+        try (NimbusClient nimbusClient = NimbusClient.getConfiguredClient(config)) {
+            return UIHelpers.makeStandardResponse(
+                    UIHelpers.getOwnerResourceSummaries(
+                            nimbusClient.getClient().getOwnerResourceSummaries(null), config),
+                    callback
+            );
+        }
+    }
+
+    /**
+     * /api/v1/owner-resources/:id -> owner resources.
+     */
+    @GET
+    @Path("/owner-resources/{id}")
+    @NimbusOp("getOwnerResourceSummaries")
```

Comment:
If someone forgets to set id, or sets it wrong, what will happen in the Authorization code when processing this NimbusOp?